tsc/frr/sonic_frr_update_process.md

```diff
@@ -0,0 +1,78 @@
+# **SONiC Community FRR Upgrade Process Proposal**
+
+FRR upgrade and patch cadence is largely on need basis in the current SONiC release program. This SONiC FRR upgrade process proposal is to formalize the FRR upgrade cadence and process for future SONiC release programs.  
+
+# SONiC FRR Version Upgrade History
+
+<p align=center>
+<img src="frr.png" alt="">
+</p>
+
+
+# FRR Project Release Cadence
+-   FRR release numbering scheme x.y.z-s#
+-   New FRR releases roughly every 4 months
+	- Apr 2023 - Release 8.5.1 (48 fixes)
+	- Mar 2023 - Release 8.5 (947+ commits)
+	- Jan 2023 - Release 8.4.2 (22 fixes)
+	- Nov 2022 - Release 8.4.1 (16 fixes)
+	- Nov 2022 - Release 8.4 (700+ commits)
+	- Aug 2022 - Release 8.3.1 (14 fixes)
+	- Jul 2022 - Release 8.3 (1000+ commits)
+	- Mar 2022 - Release 8.2.2 (800+ commits)
+	- Nov 2021 - Release 8.1.0 (1200+ commits)
+	- Jul 2021 - Release 8.0.0 (2200+ commits)
+
+-   SONiC to stay out from major/minor releases (x.y) and use patch release (.z) for stability (eg, FRR 8.3.1 instead of 8.3 if it is for 202211 release)
+
+# SONiC Release FRR Upgrade
+-   SONiC default to rebase FRR in every November community release
+-   SONiC FRR upgrade test requirements
+	-   MANDATORY: Pass all Azure pipeline build test and LGTM as required by the standard code PR merge process 
+  -   OPTIONAL: Additional tests in respect to specific changeset in the upgrade as deem necessary, manual tests should be automated and submitted to improve future test coverage   
+-   Rotate SONiC FRR maintenance duty among repo maintainer org and others (BRCM, MSFT, Alibaba, NVDA, DELL)
+-   Responsibility of SONiC FRR release maintainer
+	-   Default 12 months assignment (or until the next upgrade)
+	-   Upgrade FRR version in Nov release, resolve SONiC FRR upgrade integration issues
+  -   If there is a need for May release upgrade due to feature requirement or other reasons, it should be requested in the May release plan and have the same maintainer to handle the second upgrade
```

Comment:
I am not sure if everyone is on board with this. If for some reason a new feature is integrated in may release by a different contributor, and the different contributor asks for FRR upgrade, it becomes an unplanned activity for the maintainer. Additionally all the patches that are required for newer version plus testing is only familiar to the contributor of the feature. Requesting maintainer to integrate the patches and also qualify it is not ideal.
I would recommend that in such scenarios the effort of migrating the FRR to a newer version will be done by the new requesting contributor, since they can test, port patches and also verify their use case. However following that the maintainer for the FRR will continue to maintain the version, triage issues, etc until November.

Reply:
OK, will update the process for mid cycle upgrade